pkg/annotation/k8s.go

```diff
@@ -3,82 +3,129 @@
 
 package annotation
 
+import "sigs.k8s.io/controller-runtime/pkg/client"
+
 const (
 	// Prefix is the common prefix for all annotations
 	Prefix = "io.cilium"
 
-	// Name is an optional annotation to the NetworkPolicy
+	// ConfigPrefix is the common prefix for configuration related annotations.
+	ConfigPrefix = "config.cilium.io"
+
+	// IngressPrefix is the common prefix for ingress related annotations.
+	IngressPrefix = "ingress.cilium.io"
+
+	// NetworkPrefix is the common prefix for network related annotations.
+	NetworkPrefix = "network.cilium.io"
+
+	// PolicyPrefix is the common prefix for policy related annotations.
+	PolicyPrefix = "policy.cilium.io"
+
+	// ServicePrefix is the common prefix for service related annotations.
+	ServicePrefix = "service.cilium.io"
+
+	// PolicyName / PolicyNameAlias is an optional annotation to the NetworkPolicy
 	// resource which specifies the name of the policy node to which all
 	// rules should be applied to.
-	Name = Prefix + ".name"
+	PolicyName      = PolicyPrefix + "/name"
+	PolicyNameAlias = Prefix + ".name"
```

Comment:
I wonder whether this annotation could be dropped, since it seems not to be documented anywhere.

Reply:
Cc @christarazi

Reply:
Policy labels apply to K8s NetPol, see `pkg/k8s/network_policy.go`.

Reply:
I've updated that usage to consider both the new annotation and the alias. My doubt arose because it seemed an undocumented feature, and I didn't know whether that was intentional.